### PR TITLE
fix(shortint): fix compression encoding change not being taken into account

### PR DESCRIPTION
**Pull request by IceTDrinker**

### CODE REVIEW

**Comment by agnesLeroy** [comment]
@IceTDrinker is this something we also have to take into account for the GPU compression?

**Comment by IceTDrinker** [comment]
> @IceTDrinker is this something we also have to take into account for the GPU compression?

Oh yes sorry, forgot about the GPU implem, though to be clear the pfail is already super low with the bug, but for now I would say it’s better to update the encoding to have an even lower pfail at compression time

**Review by tmontaigu — COMMENTED** [review]
looks good

File: tfhe/src/shortint/list_compression/compression.rs [hunk 1]
```diff
@@ -131,13 +131,28 @@ impl DecompressionKey {
             return None;
         }
 
-        let carry_extract = generate_lookup_table(
+        let encryption_cleartext_modulus = packed.message_modulus.0 * packed.carry_modulus.0;
+        // We multiply by message_modulus during compression so the actual modulus for the
+        // compression is smaller
+        let compression_cleartext_modulus = encryption_cleartext_modulus / packed.message_modulus.0;
```
[the remark this answers is not in the document]
Comment (IceTDrinker) [reply tfhe/src/shortint/list_compression/compression.rs:137]: I will assert, will be safer, only thing is I mapped strictly what is done today unsure for the multiplication by the carry vs msg mod in the general case

[the remark this answers is not in the document]
Comment (IceTDrinker) [reply tfhe/src/shortint/list_compression/compression.rs:137]: I will keep the computation and add the assert

[the remark this answers is not in the document]
Comment (IceTDrinker) [reply tfhe/src/shortint/list_compression/compression.rs:141]: went for an error message

**Review by mayeul-zama — APPROVED** [review]
Thanks!

**Pull request merged**